### PR TITLE
WAL-264 Support os environment variables

### PR DESCRIPTION
**Pull request by Matrix278**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,12 +17,13 @@ There is test coverage and CI for both linuxish and windows environments, but I
...
 ```shell
-go get github.com/joho/godotenv
+go get github.com/wallester/godotenv
 ```
 
 or if you want to use it as a bin command
+
 ```shell
-go get github.com/joho/godotenv/cmd/godotenv
+go get github.com/joho/wallester/cmd/godotenv
```
Comment (Oxxyg33n) [inline README.md:26]:
```suggestion
go get github.com/wallester/godotenv/cmd/godotenv
```

File: README.md [hunk 2]
```diff
@@ -165,7 +166,7 @@ content, err := godotenv.Marshal(env)
 
 Contributions are most welcome! The parser itself is pretty stupidly naive and I wouldn't be surprised if it breaks with edge cases.
 
-*code changes without tests will not be accepted*
+_code changes without tests will not be accepted_
```
Comment (Oxxyg33n) [inline README.md:169]:
```suggestion
*code changes without tests will not be accepted*
```

File: godotenv.go [hunk 3]
```diff
@@ -317,6 +317,16 @@ func parseValue(value string, envMap map[string]string) string {
...
+	// expand variables
+	value = os.Expand(value, func(key string) string {
+		if val, ok := envMap[key]; ok {
+			return val
+		}
+		if val, ok := os.LookupEnv(key); ok {
+			return val
+		}
+		return os.Getenv(key)
+	})
```
Comment (Oxxyg33n) [inline godotenv.go:329]: Let's add newline after this one

File: godotenv.go [hunk 4]
```diff
@@ -332,7 +342,10 @@ func expandVariables(v string, m map[string]string) string {
 		if submatch[1] == "\\" || submatch[2] == "(" {
 			return submatch[0][1:]
 		} else if submatch[4] != "" {
-			return m[submatch[4]]
+			if val, ok := m[submatch[4]]; ok {
+				return val
+			}
```
Comment (Oxxyg33n) [inline godotenv.go:347]: Newline after this one

File: godotenv.go [hunk 5]
```diff
@@ -332,7 +342,10 @@ func expandVariables(v string, m map[string]string) string {
...
+			return os.Getenv(submatch[4])
 		}
```
Comment (Oxxyg33n) [inline godotenv.go:349]: & also after this one :)

```suggestion
		}
```

File: godotenv.go [hunk 6]
```diff
@@ -317,6 +317,17 @@ func parseValue(value string, envMap map[string]string) string {
...
 	}
 
+	// expand variables
+	value = os.Expand(value, func(key string) string {
+		if val, ok := envMap[key]; ok {
+			return val
+		}
```
Comment (Oxxyg33n) [inline godotenv.go:324]: Sorry, missed these, let's add a newline after this one

File: godotenv.go [hunk 7]
```diff
@@ -317,6 +317,17 @@ func parseValue(value string, envMap map[string]string) string {
...
+		if val, ok := os.LookupEnv(key); ok {
+			return val
+		}
```
Comment (Oxxyg33n) [inline godotenv.go:327]: And after this one please
